### DISCUSSION TITLE
map videos in .AVI won't load, is that expected?

### DISCUSSION THREAD

**Discussion by Quills245** [ppy/osu#25604] [Q&A] [answered]
So, idk if that's expected behavior or not so that's why i'm opening this instead of an issue, I tried to download this map on lazer https://osu.ppy.sh/beatmapsets/391867 and everything on it works normally except for the video, i made sure the option was enabled, re-downloaded the map multiple times, but it would always behave like if it had no video, so i had an idea, i went and opened the map's .osz file and saw that the video on it was on the .AVI format, so i thought maybe lazer is having a problem with that, i went and converted the file to .mp4 and adapted the map files too for them to still recognize the video and it worked! So the problem really seems to be the format

Here the edited map file (i compressed it on a .zip cuz github doesn't accept osz files):
[Are we there yet vid fix.zip](https://github.com/ppy/osu/files/13505703/Are.we.there.yet.vid.fix.zip)

Maybe this is known/expected already, if so I'm sorry for the useless discussion, i just wanted to be sure.

Os: endeavourOS
Game version: 2023.1114.1

**Comment by peppy** [comment]
@FreezyLemon may be another missing codec 😅.

**Reply by Quills245** [reply]
Yeah that's what I'd expect too

**Reply by FreezyLemon** [reply, marked as answer]
Yes, it's a missing codec. `msmpeg4v2` to be exact. I'll add it and PR.

**Reply by greyCG** [reply]
This also happens with beatmaps that use the vp6f codec (it throws a missing codec exception).
Somewhat related, there's a beatmap (https://osu.ppy.sh/beatmapsets/151#osu/337) which uses the wmv2 codec, where avformat fails to open the file.

**Reply by FreezyLemon** [reply]
> This also happens with beatmaps that use the vp6f codec (it throws a missing codec exception).

Do you have an example of a beatmap with vp6f video that doesn't work? That would make it easier to fix the problem.

I'll take a look at that wmv2 video for now.

**Reply by greyCG** [reply]
Here's the beatmap https://osu.ppy.sh/beatmapsets/7124#osu/31485
Both debugging the game and running ffprobe on the video show vp6f as the codec, and playing the beatmap with osu connected to a terminal gives this error.
```
[flv @ 0x7f72d4017b00] Could not find codec parameters for stream 0 (Video: vp6f, none, 455 kb/s): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
[flv @ 0x7f72d4017b00] Could not find codec parameters for stream 1 (Audio: mp3, 44100 Hz, stereo, 115 kb/s): unspecified frame size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
```
Here's my runtime.log after playing both beatmaps.
[runtime.log](https://github.com/ppy/osu/files/13540510/runtime.log)
